truenas/webui
# NAS-124802 / 24.04 / Add Metadata or Dedup VDEV types

**Testing**

Have at least 12 disks. Ensure it's possible to create a pool specified in the ticket. 

I. **Layout** UI element and **Width,** **VDEVs** options under **Metadata** or **Dedup** sections

1. On **Storage** \> **Create Pool**, at **Data** expandable section, select these options:
   1. Layout: **RAIDZ1**
   2. Width: **5**
   3. VDEVs: **2**
2. At **Metadata** expandable section, try to select these options:
   1. Layout: **Mirror**
   2. Width: **2**
   3. VDEVs: **1**
   * **Expected:**
     * **Layout:** this UI element is shown,
     * **Width:** list of available options should contain the proper options
     * **VDEVs:** list of available options should contain the proper options
     * The submission of the whole **Create Pool** form: does not require force-creation, and is correctly recognized after creation without raising any errors (assuming that redundancy level is equal)
3. Same for **Dedup** expandable section

II. Double check **Manual Disk Selection** adds blocks of correct layout, when special layout is chosen for **Metadata** and **Dedup** sections

1. On **Storage** > **Create Pool**, at **Data** expandable section, select same options as in "I"
2. At **Metadata** expandable section click **Manual Disk Selection** button
3. Click **Add** button in the top right corner of **Manual Selection** modal window

   **Expected:** new "MIRROR" block appears

III. Double check **Add To Pool** works

1. On **Storage** \> **Create Pool**, at **Data** expandable section, select same options as in "I"
2. Submit pool creation and wait for pool to be created
3. Add 2 disks as **Metadata** (layout MIRROR, 5 wide, 2 VDEVs)

   **Expected:** it can be done
4. Same for **Dedup**
